azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/SubnetApiMockTest.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.features;
+
+import com.squareup.okhttp.mockwebserver.RecordedRequest;
+import org.jclouds.azurecompute.arm.domain.Subnet;
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiMockTest;
+import org.testng.annotations.Test;
+import java.util.List;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertTrue;
+
+@Test(groups = "unit", testName = "SubnetApiMockTest", singleThreaded = true)
+public class SubnetApiMockTest extends BaseAzureComputeApiMockTest {
+
+    private final String subscriptionid = "SUBSCRIPTIONID";
+    private final String resourcegroup = "myresourcegroup";
+    private final String virtualNetwork = "myvirtualnetwork";
+    private final String subnetName = "mysubnet";
+    private final String apiVersion = "api-version=2015-06-15";
+
+    public void createSubnet() throws InterruptedException {
+
+        server.enqueue(jsonResponse("/createsubnetresponse.json").setResponseCode(200));
+
+        final SubnetApi subnetApi = api.getSubnetApi(resourcegroup, virtualNetwork);
+
+        Subnet.SubnetProperties properties = Subnet.SubnetProperties.create(null, "10.2.0.0/24", null);
+
+        Subnet subnet = subnetApi.createOrUpdateSubnet(subnetName, properties);
+
+        String path = String.format("/subscriptions/%s/resourcegroups/%s/providers/Microsoft.Network/virtualNetworks/%s/subnets/%s?%s", subscriptionid, resourcegroup, virtualNetwork, subnetName, apiVersion);
+        String json = String.format("{ \"properties\":{\"addressPrefix\":\"%s\"}}", "10.2.0.0/24");
```

Comment:
[minor] There is no need to Stirng.format to build this String, already using a hardcoded value.